backend/db.py

```diff
@@ -95,19 +95,24 @@ def get_all_subscribers(self):
         collection = self.db["subscribers"]
         return list(collection.find())
 
-    def add_subscriber(self, phone_number):
+    def add_subscriber(self, phone_number) -> bool:
         if not self.is_connected:
             logging.warning("not connected to database")
-            return
+            return False
         collection = self.db["subscribers"]
-        collection.insert_one({"phone_number": phone_number})
+        response = collection.update_one({"phone_number": phone_number}, {"upsert": True})
+
+        return response.modifiedCount > 0
```

Comment:
Firstly, it's `modified_count`, and secondly, this doesn't seem to work because it's only > 1 if there's an existing entry which is changed. I think `response.matched_count == 0` should give the desired behaviour.

Reply:
Yep, that sounds more logical 💯